### PR TITLE
#5235: Awaiting activities not cleared on saving

### PR DESCRIPTION
**Pull request by jtkech**
By clearing WorkflowRecords in the submit.Save controller action, the related workflows and their awaiting activities are cleared

### CODE REVIEW

**Comment by sebastienros** [comment]
Please take the comments in the issue into account. Thanks.

**Comment by jtkech** [comment]
Oh yes, but, when editing a worflow definition, there is no info about the associated workflows in the current view model. Anyway, I think we need to see if they are running instances at the time the user clicks on the "Save" button

So, to get this needed sync info on "Save", I've added an ajax get request that targets a new controller action. But not sure I've done it the right Orchard way, with right names... Let me know

File: src/Orchard.Web/Modules/Orchard.Workflows/Controllers/AdminController.cs [hunk 1]
```diff
@@ -159,6 +159,12 @@ IFormManager formManager
             return RedirectToAction("Edit", new { workflowDefinitionRecord.Id });
         }
 
+        public JsonResult State(int? id) {
+            var workflowDefinitionRecord = id.HasValue ? _workflowDefinitionRecords.Get(id.Value) : null;
+            var isRunning = workflowDefinitionRecord != null ? workflowDefinitionRecord.WorkflowRecords.Any() : false;
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Workflows/Controllers/AdminController.cs]: Can be:

`var isRunning = workflowDefinitionRecord != null && workflowDefinitionRecord.WorkflowRecords.Any()`

File: src/Orchard.Web/Modules/Orchard.Workflows/Controllers/AdminController.cs [hunk 2]
```diff
@@ -159,6 +159,12 @@ IFormManager formManager
             return RedirectToAction("Edit", new { workflowDefinitionRecord.Id });
         }
 
+        public JsonResult State(int? id) {
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Workflows/Controllers/AdminController.cs]: If it's a GET then we need to check for the permissions:
`if (!Services.Authorizer.Authorize(StandardPermissions.SiteOwner, T("Not authorized to edit workflows")))
                 return new HttpUnauthorizedResult();`

**Comment by jtkech** [comment]
Done, but I can't return an HttpUnauthorizedResult(), not compatible to a JsonResult. So I've used an AuthenticationException(). Note: If you want I can use an [HttpPost]

**Pull request merged**